packages/rspack/src/config/types.ts

```diff
@@ -1312,7 +1312,15 @@ export type ExternalItemValue =
 	| string
 	| boolean
 	| string[]
-	| Record<string, string | string[]>;
+	| {
+			/**
+			 * only available when libraryTarget and externalsType is 'umd'
+			 */
+			root: string | string[];
+			commonjs: string | string[];
+			commonjs2: string | string[];
+			amd?: string | string[];
```

Comment:
How come amd is optional?
Isn't it required to construct a real UMD import?

Reply:
After considering it, I think you are right. The `amd` field should be handled just like `commonjs` and `commonjs2`. I will fix it.